### DISCUSSION TITLE
Do data scientist and machine learning engineers need to know system design.....?

### DISCUSSION THREAD

**Discussion by GAuravY19** [mrdbourke/pytorch-deep-learning#660] [General]
Hi @mrdbourke, 

Recently, I have been around with some articles which stated system design is an integral part of machine learning whereas certain stated opposite to it.

Will you please clear me with this confusion over this....

**`Does Data Scientists and Machine learning engineers need to know system design concepts? `**

**Comment by mrdbourke** [comment]
Hi @GAuravY19 ,

Excellent question!

I'd say that system design is important in any kind of engineering role.

That includes for data science and machine learning.

Knowing what you're creating and what kind of system that will be involved in is important.

As in, say you're developing a machine learning model, knowing the system where it will run is very important.

How much compute does it need?

Where will the data be saved?

How should someone interact with the model?

How does the model and data change over time?

These are a handful of important questions to ask.

For more on machine learning system design, I'd recommend Chip Huyen's book, [Designing Machine Learning Systems](https://www.amazon.com.au/Designing-Machine-Learning-Systems-Production-Ready/dp/1098107969).

As well as her article on the same topic: https://huyenchip.com/machine-learning-systems-design/design-a-machine-learning-system.html

**Reply by GAuravY19** [reply]
Thanks a lot @mrdbourke  for clearing my confusion.
